File: 프로그래머스/1/131705. 삼총사/삼총사.cs
using System;

public class Solution {
    private int answer = 0;
    public int solution(int[] number) {
        BackTracking(0, 0, 0, number);
        return answer;
    }
    
    private void BackTracking(int start, int depth, int sum, int[] number)
    {
        if(depth == 3)
        {
            if(sum == 0)
            {
                answer++;
                return;
            }
        }
        for(int i = start; i < number.Length; i++)
        {
            sum += number[i];
            BackTracking(i + 1, depth + 1, sum, number);
            sum -= number[i];
        }        
    }
}

File: 프로그래머스/1/131705. 삼총사/삼총사.cpp
#include <string>
#include <vector>

using namespace std;

int ret = 0;
vector<int> numbers;
void BackTracking(int start, int depth, int sum)
{
    if(depth == 3)
    {
        if(sum == 0) ret++;
        return;
    }
    for(int i = start; i < numbers.size(); i++)
    {
        BackTracking(i + 1, depth + 1, sum + numbers[i]);
    }
}
int solution(vector<int> number) {
    numbers = number;
    BackTracking(0, 0, 0);
    return ret;
}